text/0000-natvis.md

```diff
@@ -0,0 +1,486 @@
+  +-- src
+      +-- main.rs
+```
+
+Where `main.rs` contains:
+
+```rust
+/// A rectangle in first quadrant
+struct FancyRect {
+    pub x: f32,
+    pub y: f32,
+    pub dx: f32,
+    pub dy: f32,
+}
+
+fn main() {
+    let mut fancy_rect = FancyRect::new(10.0, 10.0, 5.0, 5.0);
+    println!("FancyRect: {:?}", fancy_rect);
+}
+```
+
+and `Foo.natvis` contains:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+    <Type Name="foo::FancyRect">
+      <DisplayString>({x},{y}) + ({dx}, {dy})</DisplayString>
+      <Expand>
+        <Synthetic Name="LowerLeft">
+          <DisplayString>({x}, {y})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="UpperLeft">
+          <DisplayString>({x}, {y + dy})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="UpperRight">
+          <DisplayString>({x + dx}, {y + dy})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="LowerRight">
+          <DisplayString>({x + dx}, {y})</DisplayString>
+        </Synthetic>
+      </Expand>
+    </Type>
+</AutoVisualizer>
+```
+
+When viewed under WinDbg, the `fancy_rect` variable would be shown as follows:
+
+```text
+> Variables:
+  > fancy_rect: (10, 10) + (5, 5)
+    > LowerLeft: (10, 10)
+    > UpperLeft: (10, 15)
+    > UpperRight: (15, 15)
+    > LowerRight: (15, 10)
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+In rustc, a new `-Z natvis={comma-separated list of .natvis files}` flag
+will be added which instructs the compiler to take the set of .natvis
+files for a given crate and store them in the crate metadata. When running the
+linker, using the `MSVC` toolchain, a `/NATVIS` linker option would be set
+for each `.natvis` file. This includes `.natvis` files from all crate dependencies,
+if any exist, as well as the current crate and embed them into the PDB.
+
+The MSVC linker supports embedding debugger visualizations defined in a Natvis file
+(`.natvis`) into a PDB generated by LINK through the use of the `/NATVIS` linker flag.
+Since the `MSVC` linker is the only one that supports embedding Natvis files
+into a PDB, this feature would be specific to the `MSVC` toolchain only.
+
+Cargo would add new syntax in `Cargo.toml` for specifying the list of Natvis
+files to be added to the crate. The new manifest key, `natvis` would be added
+to a new `[debug-visualizations]` section. This would be in control of setting
+the `-Z natvis` flag in rustc.
+
+For example:
+
+`Cargo.toml`:
+
+```toml
+cargo-features = ["natvis"]
+
+[package]
+name = "natvis"
+version = "0.1.0"
+edition = "2018"
+
+[debug-visualizations]
+natvis = ["a.natvis", "b.natvis"]
+```
+
+This would generate a call to rustc similar to the following,
+(for simplicity purposes, most of the rustc command line has been removed):
+
+`rustc -Z natvis=path/to/file/a.natvis,path/to/file/b.natvis`
```

Comment:
This should address quoting: what happens with paths containing spaces or commas?

Reply:
Thanks for the review. I'll update the RFC to detail quoting of paths with spaces or commas.

Reply:
I've added a note that paths containing spaces will need to be quoted. Since the natvis flag takes a comma-separated list of files, I've also noted that any commas within a path would be invalid in this context.